server/keyshare/myirmaserver/memorydb.go

```diff
@@ -0,0 +1,223 @@
+package myirmaserver
+
+import (
+	"sync"
+	"time"
+
+	"github.com/privacybydesign/irmago/server/keyshare"
+)
+
+type memoryUserData struct {
+	id         int64
+	email      []string
+	logEntries []logEntry
+	lastActive time.Time
+}
+
+type memoryDB struct {
+	lock     sync.Mutex
+	userData map[string]memoryUserData
+
+	loginEmailTokens  map[string]string
+	verifyEmailTokens map[string]int64
+}
+
+func newMemoryDB() db {
+	return &memoryDB{
+		userData:          map[string]memoryUserData{},
+		loginEmailTokens:  map[string]string{},
+		verifyEmailTokens: map[string]int64{},
+	}
+}
+
+func (db *memoryDB) userIDByUsername(username string) (int64, error) {
+	db.lock.Lock()
+	defer db.lock.Unlock()
+	data, ok := db.userData[username]
+	if !ok {
+		return 0, keyshare.ErrUserNotFound
+	}
+	return data.id, nil
+}
+
+func (db *memoryDB) scheduleUserRemoval(id int64, _ time.Duration) error {
+	db.lock.Lock()
+	defer db.lock.Unlock()
+	for username, user := range db.userData {
+		if user.id == id {
+			delete(db.userData, username)
+			return nil
+		}
+	}
+	return keyshare.ErrUserNotFound
```

Comment:
* For this memory DB, I think this is indeed fine.
* For the postgres DB, there is a test invocation of this function, but it is not checked that afterwards the user still exists but is scheduled for the deletion; I will add that.
* The `tasks` component has a test that inserts a user that is scheduled for deletion, and then checks that the user cleanup task correctly deletes the user.

Reply:
I saw you added a unit test case to test whether `DeleteInProgress`, so that's a nice improvement.

Now I look to the code again, I realise that the `delete_on` of users is only acted up by the tasks component. So, on closer inspection, I apparently was looking for an integration test between the tasks component and the myirmaserver component. That would be a nice-to-have to add at some time. That is maybe not essential right now.